Formula/llvm.rb

```diff
@@ -188,17 +226,36 @@ def install
     args = %w[
       -DLLVM_OPTIMIZED_TABLEGEN=On
       -DLLVM_BUILD_LLVM_DYLIB=On
+      -DLLVM_TARGETS_TO_BUILD=X86
     ]
 
+    args << "-DLLVM_BUILD_EXTERNAL_COMPILER_RT=On" if build.with? "compiler-rt"
+    if build.with? "test"
+      args << "-DLLVM_BUILD_TESTS=On"
+      args << "-DLLVM_ABI_BREAKING_CHECKS=On"
+    end
     args << "-DLLVM_ENABLE_RTTI=On" if build.with? "rtti"
     args << "-DLLVM_INSTALL_UTILS=On" if build.with? "utils"
+    args << "-DLLVM_ENABLE_LIBCXX=On" if build.with? "libcxx" || (build.with? "clang" && !MacOS::CLT.installed?)
+    args << "-DLLVM_ENABLE_LIBCXXABI=On" if build.with? "libcxxabi"
+    args << "-DLLVM_ENABLE_DOXYGEN=On" if build.with? "doxygen"
+    args << "-DBUILD_SHARED_LIBS=On" if build.with? "shared-libs" # for developers
```

Comment:
Sorry, didn't notice this comment before. I shared my thoughts on this in https://github.com/Homebrew/homebrew-core/pull/1593#discussion_r65714184.